### PR TITLE
Architecture refactor

### PR DESCRIPTION
**Pull request by Isafagioli-Tenchi**
Architecture refactor separating the typer apps

### CODE REVIEW

**Review by edpin — COMMENTED** [review]
Need a PR description.

File: src/bin/account.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Account App
+###################################################
+
+app = typer.Typer()
+
+
+@app.command(name="me")
+def account_me():
+    """
+    Returns the details of the user account that owns the API key used by this Connection instance as per
```
Comment (edpin) [inline src/bin/account.py:17]: as per...?

File: src/bin/alerts.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+            x.value
+            for x in AlertState
+            if x != AlertState.CLOSED and x != AlertState.ACTIVE
+        ],
+        help="Only list alerts in the specified states.",
+        case_sensitive=False,
+    ),
+    severity: Optional[List[AlertSeverity]] = typer.Option(
+        [x.value for x in AlertSeverity],
+        help="Only list alerts with the specified" "severities",
```
Comment (edpin) [inline src/bin/alerts.py:41]: Why open and close quotes here? Make it one full quote.

File: src/bin/alerts.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+from src.lib.utils import output_iterable
+
+app = typer.Typer()
+
+
+@app.command(name="list")
+def alert_list(
+    organization_id: UUID = typer.Argument(..., help="UUID of the organization"),
+    scan_target_id: Optional[List[UUID]] = typer.Option(
+        None, help="Only list alerts from the specified" "scan targets."
```
Comment (edpin) [inline src/bin/alerts.py:28]: Why open and close quotes?
Also, no need for the final period (".") to be consistent with others.

File: src/bin/alerts.py [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+    ),
+    states: Optional[List[AlertState]] = typer.Option(
+        [
+            x.value
+            for x in AlertState
+            if x != AlertState.CLOSED and x != AlertState.ACTIVE
+        ],
+        help="Only list alerts in the specified states.",
```
Comment (edpin) [inline src/bin/alerts.py:36]: No period.

File: src/bin/alerts.py [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+            order=order,
+        )
+    )
+
+
+@app.command(name="list_following")
+def alert_following_list(
+    organization_id: UUID = typer.Argument(..., help="UUID of the organization"),
+    following_ids: Optional[List[UUID]] = typer.Option(
+        None, help="Only list alerts from the specified" " scan targets."
```
Comment (edpin) [inline src/bin/alerts.py:93]: No period. Open/close quotes?

File: src/bin/alerts.py [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+    ),
+    states: Optional[List[AlertState]] = typer.Option(
+        [
+            x.value
+            for x in AlertState
+            if x != AlertState.CLOSED and x != AlertState.ACTIVE
+        ],
+        help="Only list alerts in the specified states.",
```
Comment (edpin) [inline src/bin/alerts.py:101]: No period.

File: src/bin/alerts.py [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+            order=order,
+        )
+    )
+
+
+@app.command(name="list_history")
+def alert_history_list(
+    organization_id: UUID = typer.Argument(..., help="UUID of the organization"),
+    scan_target_id: Optional[List[UUID]] = typer.Option(
+        None, help="Only list alerts from the specified" "scan targets."
```
Comment (edpin) [inline src/bin/alerts.py:152]: ".", double quotes.

File: src/bin/alerts.py [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+                cursor=cursor,
+            )
+        )
+
+
+@app.command(name="list_history_following")
+def alert_history_following_list(
+    organization_id: UUID = typer.Argument(..., help="UUID of the organization"),
+    following_ids: Optional[List[UUID]] = typer.Option(
+        None, help="Only list alerts from the specified" "scan targets."
```
Comment (edpin) [inline src/bin/alerts.py:185]: ditto

File: src/bin/alerts.py [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+    ),
+    cursor: Optional[str] = typer.Option(None, help="Cursor."),
```
Comment (edpin) [inline src/bin/alerts.py:187]: Better description: "Cursor for pagination"

File: src/bin/alerts.py [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+    persist: Optional[bool] = typer.Option(False, help="Persist."),
```
Comment (edpin) [inline src/bin/alerts.py:188]: Need better description. Not sure what this is though.

File: src/bin/alerts.py [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+        ..., help="UUID of the scan target associated with the alert"
+    ),
+    alert_id: UUID = typer.Argument(..., help="UUID of the alert"),
+    state: Optional[AlertStateSetable] = typer.Option(None, help="New alert state"),
+    labels: Optional[List[str]] = typer.Option(
+        None, help="Custom label(s) for the alert"
+    ),
+    comment: Optional[str] = typer.Option(
+        None,
+        help="A comment when closing the alert with RISK_ACCEPTED, FALSE_POSITIVE, MITIGATING_CONTROL",
```
Comment (edpin) [inline src/bin/alerts.py:319]: Not "closing", "setting the alert state to" instead.

File: src/bin/scan_target_groups.py [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+
+@app.command(name="insert")
+def scan_target_groups_insert(
+    organization_id: UUID = typer.Argument(..., help="UUID of the organization"),
+    scan_target_group_id: UUID = typer.Argument(
+        ..., help="UUID of the scan target group"
+    ),
+    region: str = typer.Argument(..., help="Oracle cloud region"),
+    tenancy_id: str = typer.Argument(..., help="Oracle tenancyId"),
+    user_id: str = typer.Argument(..., help="Oracle UserId"),
```
Comment (edpin) [inline src/bin/scan_target_groups.py:132]: Please double check whether it's UserId or userId like tenancyId. Or perhaps it's TenancyId. It seems strange as it is, check on the UI and let's standardize.

File: src/bin/summary.py [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+    dump_json(client.get_alert_summaries(organization_id, scan_target_id))
+
+
+@app.command(name="alert_following")
+def summary_alert_following(
+    organization_id: UUID = typer.Argument(..., help="UUID of the organization"),
+    following_ids: Optional[List[UUID]] = typer.Option(
+        None,
+        help="Only summarize alerts from the" "specified following, defaults to" "all.",
```
Comment (edpin) [inline src/bin/summary.py:35]: "." and open/close quotes.

File: test/account_test.py [hunk 14]
```diff
@@ -0,0 +1,6 @@
+import unittest
+
+
+class AccountTest(unittest.TestCase):
+    def test_account_me(self):
+        pass
```
Comment (edpin) [inline test/account_test.py:6]: No, this is not good. Either write the test now or don't create this harness. This leads to a false sense of safety when running tests, where one thinks they're passing, but in fact, there are no tests.

File: test/alerts_test.py [hunk 15]
```diff
@@ -0,0 +1,24 @@
+import unittest
+
+
+class AlertsTest(unittest.TestCase):
```
Comment (edpin) [inline test/alerts_test.py:4]: Big no-no. See above.

**Review by edpin — CHANGES_REQUESTED** [review]
Good stuff. Lots of comments, but most are cosmetic. 

Please don't include empty tests. Leave them in a branch or locally on your git client and add them back later as you fill them in. See why below.

File: test/account_test.py (hunk 14 above)
Comment (RomuloGatto) [reply test/account_test.py:6]: @edpin  All of the pre-existent tests were just a pass, we just reorganized it now, as soon as we refactor the SDK, we will come back to fix the tests

Comment (edpin) [reply test/account_test.py:6]: Understood, but it was a bad idea to have them blank/pass in the first place. Not sure who put it there, but we can fix it now. We need to remove them and add them one-by-one later. Better no tests than a false sense of safety.

**Review by edpin — APPROVED** [review]

**Review by RomuloGatto — APPROVED** [review]

**Pull request merged**